registry/server/tagstorage_zk.go

```diff
@@ -166,14 +167,70 @@ func (t *ZKTagStorage) GetTags(o KafkaObject) (TagSet, error) {
 	return tags, nil
 }
 
-// DeleteTags deletes all tags in the Tags for the requested KafkaObject.
-func (t *ZKTagStorage) DeleteTags(o KafkaObject, ts Tags) error {
+// GetAllTags returns all tags stored in the tagstore, keyed by the resource they correspond to.
+func (t *ZKTagStorage) GetAllTags() (map[KafkaObject]TagSet, error) {
+
+	tags := map[KafkaObject]TagSet{}
+
+	brokerTags, err := t.GetAllTagsForType("broker")
+	if err != nil {
+		return nil, err
+	}
+	for broker, brokerTags := range brokerTags {
+		tags[broker] = brokerTags
+	}
+
+	topicTags, err := t.GetAllTagsForType("topic")
+	if err != nil {
+		return nil, err
+	}
+	for topic, topicTags := range topicTags {
+		tags[topic] = topicTags
+	}
+
+	return tags, nil
+}
+
+// GetAllTagsForType gets all the tags for objects of the given type. A convenience method that makes getting every tag a little easier.
+func (t *ZKTagStorage) GetAllTagsForType(kafkaObjectType string) (map[KafkaObject]TagSet, error) {
+	zNode := fmt.Sprintf("/%s/%s/", t.Prefix, kafkaObjectType)
+	children, err := t.ZK.Children(zNode)
+	if err != nil {
+		switch err.(type) {
+		case kafkazk.ErrNoNode:
+			return nil, ErrKafkaObjectDoesNotExist
+		default:
+			return nil, err
+		}
+	}
+
+	objectTags := map[KafkaObject]TagSet{}
+
+	for objectId := range children {
+		object := KafkaObject{Type: kafkaObjectType, ID: strconv.Itoa(objectId)}
+		tags, err := t.GetTags(object)
+		if err != nil {
+			switch err.(type) {
+			case kafkazk.ErrNoNode:
+				return nil, ErrKafkaObjectDoesNotExist
+			default:
+				return nil, err
+			}
+		}
+		objectTags[object] = tags
+	}
+
+	return objectTags, nil
+}
+
+// DeleteTags deletes all tags in the list of keys for the requested KafkaObject.
+func (t *ZKTagStorage) DeleteTags(o KafkaObject, keysToDelete []string) error {
 	// Sanity checks.
 	if !o.Complete() {
 		return ErrInvalidKafkaObjectType
 	}
 
-	if len(ts) == 0 {
+	if len(keysToDelete) == 0 {
```

Comment:
Heh yeah better naming re: our convo on "deleting Tags" vs "accepting tag keys to delete"